docs/writing-rules/pattern-syntax.mdx

```diff
@@ -515,7 +515,27 @@ When referencing an ellipsis metavariable in a rule message or [metavariable-pat
 - message: Call to foo($...ARGS)
 ```
 
-### Displaying matched metavariables in rule messages
+### Anonymous metavariables
+
+Anonymous metavariables are used to specify that there a metavariable exists in the pattern you want to capture.
+
+An anonymous metavariable always takes the form `$_`. Variables such as `$_1` or `$_2` are **not** anonymous. You can use more than one anonymous metavariable in a rule definition.
+
+For example, if you want to specify that a function should **always** have 3 arguments, then you can use anonymous metavariables:
+
+```yaml
+- pattern: def function($_, $_, $_)
+```
+
+Anonymous metavariables also communicate to the reader that their values are not relevant, but rather their occurrence in the pattern.
+
+### Metavariable unification
+
+For search mode rules, metavariables with the same name are treated as the same metavariable across and within pattern operators **except for `pattern-either`**.
+
+For taint mode rules, metavariable unification is **not** enabled by default. To enforce unification, set `taint_unify_mvars: true` under the rule `options` key. When `taint_unify_mvars: true` is set, a metavariable defined in `pattern-sinks` and `pattern-sources` with the same name is treated as the same metavariable. See [Metavariables, rule message, and unification](/writing-rules/data-flow/taint-mode#metavariables-rule-message-and-unification) for more information.
```

Comment:
technically, unification is only not enabled between the source and sink. the patterns within the source will still unify with each other, maybe worth clarifying